lib/data/dto/expenses/expense_dto.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:trip_tally/domain/entities/expenses/expense_entity.dart';
+
+part 'expense_dto.freezed.dart';
+
+part 'expense_dto.g.dart';
+
+@freezed
+class ExpenseDto with _$ExpenseDto {
+  const factory ExpenseDto({
+    required String name,
+    required String date,
+    required double amount,
+    required String currency,
+    required String tripId,
+    required String userId,
+  }) = _ExpenseDto;
+
+  factory ExpenseDto.fromJson(Map<String, dynamic> json) => _$ExpenseDtoFromJson(json);
+
+  factory ExpenseDto.fromEntity(ExpenseEntity entity, String userId) {
+    return ExpenseDto(
+      name: entity.name,
+      date: entity.date,
+      amount: entity.amount,
+      currency: entity.currency,
+      tripId: entity.tripId,
+      userId: userId,
```

Comment:
As we establish you dont need to do that as the userId is parsed from token in the backend side